### PR TITLE
Refactoring and cleanups

### PR DESCRIPTION
**Pull request by stv0g**
I streamlined the Dockerfile which is now doing a multi-stage build which greatly simplifies it and improves cache utilization.
I also fixed some Markdown issues in the README and made general code-style improvements to the shell scripts.
I enabled IPv6 support in Mongoose.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,46 +1,47 @@
...
-# build with `docker build --build-arg recorder_version=x.y.z '
-ARG recorder_version=0.8.8
+ARG RECORDER_VERSION=0.8.8
+# ARG RECORDER_VERSION=master
 
-COPY entrypoint.sh /entrypoint.sh
-COPY config.mk /config.mk
-COPY recorder.conf /etc/default/recorder.conf
-COPY recorder-health.sh /usr/local/sbin/recorder-health.sh
+RUN apk add \
```
Comment (jpmens) [inline Dockerfile:6]: I note you've removed `--no-cache`. I think somebody explained to me that option reduces the image size. Is that not (no longer) the case/

File: Dockerfile [hunk 2]
```diff
@@ -50,4 +51,4 @@ COPY JSON.lua /config/JSON.lua
 
 EXPOSE 8083
 
-ENTRYPOINT ["/entrypoint.sh"]
+ENTRYPOINT ["/usr/sbin/ot-recorder"]
```
Comment (jpmens) [inline Dockerfile:54]: by removing `entrypoint.sh` the Recorder will not initialize its data store. That won't work.

File: README.md [hunk 3]
```diff
@@ -66,28 +70,30 @@ The `/store` volume of the container is used for persistent storage of location
...
+docker run -d -p 8083:8083 -v ./store:/store owntracks/recorder
 ```
 
 If nothing is mounted at `/store`, docker will create a unique volume
 automatically. However up on recreation of the docker container, this process
 will be repeated and another unique volume will be created. As a result, the
 container will have forgotten about previous tracks.
 
-## TLS between MQTT broker and Recorder
+## TLS between MQTT Broker and Recorder
```
Comment (jpmens) [inline README.md:89]: I'm not typically a great fan of uppercased nouns in English, also not in titles, but it'll pass. :)

**Review by jpmens — COMMENTED** [review]
Thank you, Steffen, for your contribution which basically looks good to me (as far as I understand Docker :)

I do not see where `ot-recorder` initializes its back-end data store now that you've eliminated `entrypoint.sh`. Could you please verify that?

File: Dockerfile (hunk 1 above)
Comment (stv0g) [reply Dockerfile:6]: Ah okay, I see. I will add it again. It was causing issues on my side. But I think thats resolved now.

File: README.md (hunk 3 above)
Comment (stv0g) [reply README.md:89]: :D I dont have a preference here. I just noticed the inconsistency between **b**roker and **R**ecorder which triggered my OCD.

File: Dockerfile (hunk 2 above)
Comment (stv0g) [reply Dockerfile]: Done. I added it again.

File: README.md (hunk 3 above)
Comment (jpmens) [reply README.md]: Ah, yes, that's actually on purpose: I say Recorder to mean `ot-recorder` as in a bit of a proper name.
But fine; leave it as you've done.

File: recorder.conf [hunk 4]
```diff
@@ -4,7 +4,7 @@
 # and its associated utilities to override compiled-in defaults.
 
 OTR_TOPICS = "owntracks/#"
-OTR_HTTPHOST = "0.0.0.0"
+OTR_HTTPHOST = "[::]"
```
Comment (jpmens) [inline recorder.conf:7]: Sorry, I missed this on first pass.
Could you please make `0.0.0.0` default as most people will more easily recognize that as an IP address (I fear), and add a commented out IPv6 version?

Comment (stv0g) [reply recorder.conf:7]: Yes I thought about it. However, the issue is that binding on `0.0.0.0` will only listen on IPv4. Whereas binding on `[::]` will listen on IPv4 and IPv6 which I think should be the default.

How about if I add a comment with the IPv4 version?

Comment (jpmens) [reply recorder.conf:7]: How about this:

```
# binding on 0.0.0.0 will listen on IPv4 only
# binding on [::] will listen on IPv4 and IPv6
# OTR_HTTPHOST = "0.0.0.0"
OTR_HTTPHOST = "[::]"
```

or equivalent.

Comment (stv0g) [reply recorder.conf:11]: Done :)

**Comment by stv0g** [comment]
Just noticed that Travis is failing. I will fix this ASAP

**Comment by jpmens** [comment]
The build passed; I think it has something to do permissions.

**Comment by jpmens** [comment]
Merged. Thank you very much, Steffen!

**Comment by jpmens** [comment]
Image has shrunk a bit :-) 

```
REPOSITORY           TAG        IMAGE ID       CREATED         SIZE
owntracks/recorder   0.8.8-14   7556f8ad6c26   3 minutes ago   12.3MB
owntracks/recorder   latest     7556f8ad6c26   3 minutes ago   12.3MB
owntracks/recorder   0.8.8-13   1114758b7723   2 months ago    14.3MB
```

**Comment by saesh** [comment]
Hey! 

Just wanted to let you know that [this change](https://github.com/owntracks/docker-recorder/pull/52/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R59=) together with [this change](https://github.com/owntracks/docker-recorder/pull/52/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R59=) now overrides [`OTR_TOPICS` in the `recorder.conf`](https://github.com/owntracks/recorder#configuration-file=) of owntracks-recorder itself.

This was somewhat unexpected as `OTR_TOPICS` is not supported as an environment variable in owntracks-recorder and the only way to restrict the topics on the recorder was using the `recorder.conf` file. Now docker-recorder governs this behavior by an environment variable `OTR_TOPIC` (note, without 's') making the use of `OTR_TOPICS` in the `recorder.conf` obsolete when using docker-recorder. The change was not easy to spot and I had to clean up quite a bit as my instances of owntracks-recorder where collecting data from all topics all of a sudden.

**Comment by jpmens** [comment]
@saesh thanks for reporting this, and I am very sorry it caused you additional work.

To be honest, I thought the idea was good because it would at least temporarily solve the issue of `ot-recorder` not yet supporting `$OTR_TOPICS`, but I didn't think it through when accepting this pull.

Right now I'm not sure whether we should retract this feature, as we simply don't know how many people will be inconvenienced either way...

I'm open to suggestions.

**Comment by saesh** [comment]
No worries @jpmens.

I think for newcomers to the project using docker-recorder it might be unclear how to achieve certain things as some configuration is done here (restricting topics), and some are done via environment variables in owntracks-recorder and yet some other are only supported using the `recorder.conf` file.

I would suggest to make most of the important configuration available through environment variables in owntracks-recorder. docker-recorder should be responsible for one thing only: making owntracks-recorder available through Docker and delegating any set of environment variables to owntracks-recorder (better yet without any knowledge what environment variables owntracks-recorder supports). That way it does not matter if you use owntracks-recorder by itself or through docker-recorder; the environment is the same (same env variables) only the underlying runtime is different. Then people can reference the official owntracks-recorder docs and use docker-recorder transparently.

**Pull request merged**